core/src/main/scala/scalaz/std/Array.scala

```diff
@@ -0,0 +1,8 @@
+package scalaz
+package std
+
+trait ArrayInstances {
+  implicit def arrayShow[A](implicit SA: Show[A]): Show[Array[A]] = new Show[Array[A]] {
+    override def shows(a: Array[A]): String = a.map(SA.shows).mkString("[", ",", "]")
```

Comment:
The `map` call makes a temp `Array[String]`, which is probably suboptimal. Even using `.iterator` may be faster.